lightning/src/ln/channel.rs

```diff
@@ -4514,7 +4540,21 @@ impl<Signer: Sign> Channel<Signer> {
 		if self.cur_holder_commitment_transaction_number != INITIAL_COMMITMENT_NUMBER {
 			panic!("Tried to send an accept_channel for a channel that has already advanced");
 		}
+		if !self.inbound_awaiting_accept {
+			panic!("The inbound channel has already been accepted");
+		}
+
+		self.inbound_awaiting_accept = false;
+
+		return self.get_accept_channel_message();
+	}
 
+	/// This function is used to explicitly generate a [`msgs::AcceptChannel`] message for an
+	/// inbound channel. If the intention is to accept an inbound channel, you should use the
+	/// [`Channel::accept_inbound_channel`] function instead.
+	///
+	/// [`msgs::AcceptChannel`]: crate::ln::msgs::AcceptChannel
+	pub fn get_accept_channel_message(&self) -> msgs::AcceptChannel {
```

Comment:
Don't think it matters too much but we could make this pub just in testing

Reply:
Yea, I'd strongly prefer this be a private method with a `#[cfg(test)]` method that exposes it just for testing.

Reply:
Ok sure, I'll make it private and expose a function for testing!